From 5d24bcd56dc7e334cad365981da112aafbbb93a4 Mon Sep 17 00:00:00 2001
From: Kenneth Lu <kuohsianglu@gmail.com>
Date: Sat, 2 Dec 2017 17:54:13 +0800
Subject: [PATCH] ath6kl: improve endianness handling
Git-commit: 5d24bcd56dc7e334cad365981da112aafbbb93a4
Patch-mainline: v4.16-rc1
References: FATE#326906

Replace ntohs with be16_to_cpu to do endian conversions for ethhdr
h_proto assignment.

Signed-off-by: Kenneth Lu <kuohsianglu@gmail.com>
Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/ath/ath6kl/txrx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/txrx.c b/drivers/net/wireless/ath/ath6kl/txrx.c
index 1379906bf849..8da9506f8c2b 100644
--- a/drivers/net/wireless/ath/ath6kl/txrx.c
+++ b/drivers/net/wireless/ath/ath6kl/txrx.c
@@ -1001,7 +1001,7 @@ static void aggr_slice_amsdu(struct aggr_info *p_aggr,
 
 	while (amsdu_len > mac_hdr_len) {
 		hdr = (struct ethhdr *) framep;
-		payload_8023_len = ntohs(hdr->h_proto);
+		payload_8023_len = be16_to_cpu(hdr->h_proto);
 
 		if (payload_8023_len < MIN_MSDU_SUBFRAME_PAYLOAD_LEN ||
 		    payload_8023_len > MAX_MSDU_SUBFRAME_PAYLOAD_LEN) {
-- 
2.19.2

